spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/DirectedGraph.scala

```diff
@@ -148,12 +148,17 @@ class DirectedGraph[T](val source : Node[T], val reverse : Boolean = false) exte
     })
     bfs.foreach(node => {
       if (reverseEdge) {
-        node.prevNodesAndEdges.foreach(prevNodeAndEdge => {
-          oldToNew.get(node).add(oldToNew.get(prevNodeAndEdge._1), prevNodeAndEdge._2)
+        node.nextNodesAndEdges.foreach(nextNodeAndEdge => {
+          // some node may be ignored
```

Comment:
Please add more comments here. e.g.
// keep the order in the nextNodes array in the current node. As we go through all node in bfs, the prevNodes order should be preserved.
// Some next nodes may be not included in the graph